File: KnowledgeSharingApi/KnowledgeSharingApi.Domain/Models/Entities/Views/ViewLesson.cs
using KnowledgeSharingApi.Domains.Models.Entities.Entities;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace KnowledgeSharingApi.Domains.Models.Entities.Views
{
    [Table("ViewLesson")]
    public class ViewLesson : Lesson
    {
        public string Username { get; set; } = string.Empty;

        public string FullName { get; set; } = string.Empty;

        public string? Avatar { get; set; }

        public string? Cover { get; set; }


        public int? TotalStar { get; set; }

        public int? SumStar { get; set; }

        public double? AverageStar { get; set; }

        public int? TotalComment { get; set; }

        protected override ViewLesson Init()
        {
            return new ViewLesson();
        }
    }
}

File: KnowledgeSharingVue/knowledge-sharing-vue/src/js/models/views/view-lesson.js
// Framework: .NET Core
// Technology Stack: Entity Framework

import Lesson from "../entities/lesson";

class ViewLesson extends Lesson {
    constructor() {
        super();
        this.Username = null;
        this.FullName = null;
        this.Avatar = null;
        this.Cover = null;
        this.TotalStar = null;
        this.SumStar = null;
        this.AverageStar = null;
        this.TotalComment = null;
    }

    init() {
        return new ViewLesson();
    }

}

export default ViewLesson;